packages/store/src/selectors/selector-utils.ts

```diff
@@ -84,29 +40,24 @@ export function createSelector<T extends (...args: any[]) => any>(
       }
     };
   };
-
-  selectorMetaData.makeRootSelector = makeRootSelector;
-
-  return memoizedFn;
 }
 
-function setupSelectorMetadata<T extends (...args: any[]) => any>(
+export function createMemoizedSelectorFn<T extends (...args: any[]) => any>(
   originalFn: T,
-  creationMetadata: CreationMetadata | undefined
+  creationMetadata: Partial<CreationMetadata> | undefined
 ) {
-  const selectorMetaData = ensureSelectorMetadata(originalFn);
-  selectorMetaData.originalFn = originalFn;
-  let getExplicitSelectorOptions = () => ({});
-  if (creationMetadata) {
-    selectorMetaData.containerClass = creationMetadata.containerClass;
-    selectorMetaData.selectorName = creationMetadata.selectorName;
-    getExplicitSelectorOptions =
-      creationMetadata.getSelectorOptions || getExplicitSelectorOptions;
-  }
-  const selectorMetaDataClone = { ...selectorMetaData };
-  selectorMetaData.getSelectorOptions = () =>
-    getLocalSelectorOptions(selectorMetaDataClone, getExplicitSelectorOptions());
-  return selectorMetaData;
+  const containerClass = creationMetadata && creationMetadata.containerClass;
+  const wrappedFn = function wrappedSelectorFn(...args: any[]) {
+    const returnValue = originalFn.apply(containerClass, args);
+    if (returnValue instanceof Function) {
+      const innerMemoizedFn = memoize.apply(null, [returnValue]);
+      return innerMemoizedFn;
+    }
+    return returnValue;
+  } as T;
+  const memoizedFn = memoize(wrappedFn);
+  Object.setPrototypeOf(memoizedFn, originalFn);
+  return memoizedFn;
```

Comment:
Same here, can we have comments? Even me having troubles understanding it.

Reply:
I can do this in a future PR. This is old code that has just been moved, and I wouldn't want it to block this PR.